crates/sui/src/validator_commands.rs

```diff
@@ -0,0 +1,342 @@
+                    read_network_keypair_from_file(network_key_file_name)?;
+                let network_keypair: NetworkKeyPair =
+                    read_network_keypair_from_file(worker_key_file_name)?;
+                let pop =
+                    generate_proof_of_possession(&keypair, (&account_keypair.public()).into());
+                let validator_info = GenesisValidatorInfo {
+                    info: sui_config::ValidatorInfo {
+                        name,
+                        protocol_key: keypair.public().into(),
+                        worker_key: worker_keypair.public().clone(),
+                        account_key: account_keypair.public(),
+                        network_key: network_keypair.public().clone(),
+                        gas_price,
+                        commission_rate: 0,
+                        network_address: Multiaddr::try_from(format!(
+                            "/dns/{}/tcp/8080/http",
+                            host_name
+                        ))?,
+                        p2p_address: Multiaddr::try_from(format!("/dns/{}/udp/8084", host_name))?,
+                        narwhal_primary_address: Multiaddr::try_from(format!(
+                            "/dns/{}/udp/8081",
+                            host_name
+                        ))?,
+                        narwhal_worker_address: Multiaddr::try_from(format!(
+                            "/dns/{}/udp/8082",
+                            host_name
+                        ))?,
+                        description,
+                        image_url,
+                        project_url,
+                    },
+                    proof_of_possession: pop,
+                };
+                // TODO set key files permisssion
+                let validator_info_file_name = dir.join("validator.info");
+                let validator_info_bytes = serde_yaml::to_vec(&validator_info)?;
+                fs::write(validator_info_file_name.clone(), validator_info_bytes)?;
+                println!(
+                    "Generated validator info file: {:?}.",
+                    validator_info_file_name
+                );
+                SuiValidatorCommandResponse::MakeValidatorInfo
+            }
+            SuiValidatorCommand::BecomeCandidate { file } => {
+                let validator_info_bytes = fs::read(file)?;
+                // Note: we should probably rename the struct or evolve it accordingly.
+                let validator_info: GenesisValidatorInfo =
+                    serde_yaml::from_slice(&validator_info_bytes)?;
+                let validator = validator_info.info;
+
+                let args = vec![
+                    CallArg::Pure(
+                        bcs::to_bytes(&AuthorityPublicKeyBytes::from_bytes(
+                            validator.protocol_key().as_bytes(),
+                        )?)
+                        .unwrap(),
+                    ),
```

Comment:
No need to change anything...I do wonder if we need to improve our Rust SDK to better handle these types of calls.

Reply:
I tried transaction builder but sadly found SuiJsonValue very challenging to accommodate vectors of bytes. Perhaps I just don't know how to use it